x-pack/plugins/enterprise_search/public/applications/__mocks__/shallow_useeffect.mock.ts

```diff
@@ -19,3 +23,14 @@ jest.mock('react', () => ({
  *
  * // ... etc.
  */
+/**
+ * Example unmount() usage:
+ *
+ * import { onUnmount } from '../../../__mocks__/shallow_useeffect.mock';
```

Comment:
Argh so sorry, just noticed this, this one's totally an error on my part 🤦‍♀️ I was playing around with different helper names but ended up preferring `unmountHandler` (your naming) and because unmount/onUnmount implied we were actually unmounting the component (not the case). I forgot to update my example though, sorry!

```suggestion
 * import { unmountHandler } from '../../../__mocks__/shallow_useeffect.mock';
```